### ISSUE TITLE
Refactor to be event-based

### ISSUE THREAD

**Issue by bkeepers** [probot/scheduler#6]
The current usage of visitor feels weird along side scripts that are largely event-based. I was thinking about how we could start to move to a more event-based pattern, and here's an idea: 

```js
const visitor = require('probot-visitor');

module.exports = robot => {
  // How often to schedule events, and the name of the event to schedule'
  visitor(robot).schedule('daily', 'sweep');

  robot.on('scheduled.sweep', (event, context) => {
    // event.payload.repository - repository object
    // event.payload.installation - installation object
    // context.github - authenticated GitHub client
  });
};
```

cc @lee-dohm

**Comment by lee-dohm** [comment]
I like it! :100:

Maybe instead of `scheduled` we could use `visitor` as the root name? Perhaps other plugins would like to create synthesized events like this? If so, then having a by-convention namespacing system would help facilitate that.

**Comment by bkeepers** [comment]
> Maybe instead of `scheduled` we could use `visitor` as the root name?

Good point. What would you think about renaming this plugin to `probot-scheduler`? Original I called it `visitor` because it visits each repository, but I think that's an implementation detail. What it does is "scheduling". Does that make sense?

**Comment by lee-dohm** [comment]
It schedules, yes ... but I would tend to assume that something called
scheduler would be more flexible than simply "hourly", "daily", or such.
Maybe "cron"? 😉 Naming is hard ...

**Comment by bkeepers** [comment]
Eventually, I think it would allow more flexibility. For example, Lambda has [scheduled events](http://docs.aws.amazon.com/lambda/latest/dg/with-scheduled-events.html), which can be an interval (hour, day, etc) or a cron expression.

I think this plugin should offer the same.

**Comment by lee-dohm** [comment]
Sounds good! `scheduler` it is :+1:

**Comment by bkeepers** [comment]
Fixed in #7
